bot.py

```diff
@@ -432,33 +569,52 @@ def translate_slack_usernames(self, unchecked_authors):
         """
         await self.typing(channel_id)
-        if has_command(['release', 'notes'], words):
-            await self.commits_since_last_release(repo_info)
-        elif has_command(['release'], words) or has_command(['start', 'release'], words):
-            version = get_version_number(words[-1])
-
-            loop.create_task(self.delay_message(repo_info))
-            await self.do_release(repo_info, version)
-            await self.wait_for_checkboxes(repo_info)
-        elif has_command(['finish', 'release'], words):
-            await self.finish_release(repo_info)
-        elif has_command(['wait', 'for', 'checkboxes'], words):
-            await self.wait_for_checkboxes(repo_info)
-        elif has_command(['hi'], words):
-            await self.say(
-                channel_id,
-                "A Mongol army? Really? Uh, I must have had the dial set for"
-                " 'Hun.' Oh, well, you don't look a gift horde in the mouth, so... hello! "
-            )
-        elif has_command(['karma'], words):
-            start_date = parse(words[1]).date()
-            await self.karma(channel_id, start_date)
-        elif has_command(['what', 'needs', 'review'], words):
-            await self.needs_review(channel_id)
-        elif has_command(["version"], words):
-            await self.report_version(repo_info)
-        else:
-            await self.say(channel_id, "Oooopps! Invalid command format")
+        commands = self.make_commands()
+        for command, parsers, command_func, _ in commands:
+            command_words = command.split()
+            if has_command(command_words, words):
+                args = words[len(command_words):]
+                if len(args) != len(parsers):
+                    await self.say(
+                        channel_id,
+                        "Careful, careful. I expected {expected_num} words but you said {actual_num}.".format(
+                            expected_num=len(parsers),
+                            actual_num=len(args),
+                        )
+                    )
+                    return
+
+                parsed_args = []
+                for arg, parser in zip(args, parsers):
+                    try:
+                        parsed_args.append(parser.func(arg))
+                    except:  # pylint: disable=bare-except
+                        log.exception("Parser exception")
+                        await self.say(
+                            channel_id,
+                            "Oh dear! You said `{word}` but I'm having trouble figuring out what that means.".format(
+                                word=arg,
+                            )
+                        )
+                        return
+
+                await command_func(
+                    CommandArgs(
+                        repo_info=repo_info,
+                        channel_id=channel_id,
+                        args=[parser(arg) for arg, parser in zip(args, parsers)],
```

Comment:
Should this just be passing `parsed_args` instead? If not, it doesn't appear we use `parsed_args` anywhere.

Reply:
Yes, I just fixed it